File: 2015/1/pb.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main() {
    ifstream fin("input.txt");
    string line;
    getline(fin, line);
    vector<char> dir(line.begin(), line.end());

    // dir = {'(', ')', ')', '(', ')', ')', '(', ')', ')'};

    int count = 0;
    for (int i = 0; i < dir.size(); i++) {
        if (dir[i] == '(') count++;
        if (dir[i] == ')') count--;

        if (count == -1) {
            cout << i + 1 << endl;
            cout << "Count: " << count << endl;
        }
    }

    cout << count << endl;

    return 0;
}

File: 2016-py/day3/pb.py
def is_valid(a, b, c):
    return a + b > c and a + c > b and b + c > a


def main():
    triangles = []

    with open("input.txt", "r") as fin:
        for line in fin:
            triangles.append(list(map(int, line.split())))

    count = 0
    for i in range(0, len(triangles), 3):
        for triangle in zip(*triangles[i : i + 3]):
            if is_valid(*triangle):
                count += 1

    print(count)


if __name__ == "__main__":
    main()

